### PR TITLE
CORE-12218: New REST endpoint for flow result as JSON

### PR DESCRIPTION
**Pull request by vkolomeyko**
New REST endpoint introduced: `GET /api/v1/flow/{holdingidentityshorthash}/{clientrequestid}/result/`
Few Smoke tests been updated to make use of the new REST API method.

I have tested new endpoint locally using Combined Worker:
image

### CODE REVIEW

File: components/flow/flow-rest-resource-service/src/main/kotlin/net/corda/flow/rest/v1/types/response/FlowStatusResponse.kt [hunk 1]
```diff
@@ -20,7 +21,7 @@ data class FlowStatusResponse(
     val clientRequestId: String?,
     val flowId: String?,
     val flowStatus: String,
-    val flowResult: String?,
+    val flowResult: JsonObject?,
```
Comment (vkolomeyko) [inline components/flow/flow-rest-resource-service/src/main/kotlin/net/corda/flow/rest/v1/types/response/FlowStatusResponse.kt:24]: This is main part of the change.

File: processors/rest-processor/src/integrationTest/resources/swaggerBaseline.json [hunk 2]
```diff
@@ -4409,9 +4409,23 @@
             "example" : "string"
           },
           "flowResult" : {
-            "type" : "string",
```
Comment (vkolomeyko) [inline processors/rest-processor/src/integrationTest/resources/swaggerBaseline.json:4412]: It is possible to see that with change OpenAPI expanded from just `string` to `anyOf: string, number, object,... `.

**Review by ronanbrowne — DISMISSED** [review]
LGTM  (build files reviewed)

File: processors/rest-processor/src/integrationTest/resources/swaggerBaseline.json [hunk 3]
```diff
@@ -888,6 +888,54 @@
         }
       }
     },
+    "/flow/{holdingidentityshorthash}/{clientrequestid}/result" : {
```
Comment (vkolomeyko) [inline processors/rest-processor/src/integrationTest/resources/swaggerBaseline.json:891]: New REST API endpoint introduced.

**Review by mnesbit — APPROVED** [review]
Seems reasonable to me, but I have had limited time to review this

**Review by filipesoliveira — APPROVED** [review]

**Review by josephzunigadaly — APPROVED** [review]

**Review by ronanbrowne — APPROVED** [review]

**Review by mag1c-48 — APPROVED** [review]
LGTM build files

**Pull request merged**